### ISSUE TITLE
twisted/test/test_tpfile.py fails on Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8668]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8668|
|Type|enhancement|
|Created|2016-07-25 05:35:52Z|
|Branch|https://github.com/twisted/twisted/tree/8668-rodrigc-tpfile-py3|

If I try to do:

```
python3 bin/trial twisted/test/test_tpfile.py
```

I get:

```
twisted/test/test_tpfile
  py ...                                                                [ERROR]

===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 781, in loadByName
    return self.suiteFactory([self.findByName(name, recurse=recurse)])
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 682, in findByName
    return self.loadFile(_name, recurse=recurse)
  File "/Users/crodrigues/twisted_win/twisted/trial/runner.py", line 652, in loadFile
    module = SourceFileLoader(name, fileName).load_module()
  File "<frozen importlib._bootstrap_external>", line 383, in _check_name_wrapper
    
  File "<frozen importlib._bootstrap_external>", line 805, in load_module
    
  File "<frozen importlib._bootstrap_external>", line 664, in load_module
    
  File "<frozen importlib._bootstrap>", line 268, in _load_module_shim
    
  File "<frozen importlib._bootstrap>", line 693, in _load
    
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
    
  File "<frozen importlib._bootstrap_external>", line 660, in exec_module
    
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "twisted/test/test_tpfile.py", line 11, in <module>
    import StringIO
builtins.ImportError: No module named 'StringIO'

twisted/test/test_tpfile.py
-------------------------------------------------------------------------------
Ran 1 tests in 0.013s

FAILED (errors=1)
```

<details><summary>Searchable metadata</summary>

```
trac-id__8668 8668
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8668_rodrigc_tpfile_py3 8668-rodrigc-tpfile-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469424952666286 1469424952666286
changetime__1474198049187875 1474198049187875
version__None None
owner__GitHub__noreply_____ GitHub <noreply@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/408

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|GitHub set owner to GitHub<br>GitHub set status to `closed`|
|-|-|

In changeset b7f3cfc3ee270ef33f0f8060c1f3bf1c2ab6395d
```
#!CommitTicketReference repository="" revision="b7f3cfc3ee270ef33f0f8060c1f3bf1c2ab6395d"
Merge pull request #408 from twisted/8668-rodrigc-tpfile-py3

Author: rodrigc
Reviewer: alex
Fixes: #8668

Port twisted.test.test_tpfile to Python 3
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
